.github/actions/composite/setupNode/action.yml

```diff
@@ -0,0 +1,20 @@
+name: Set up Node
+description: Set up Node
+
+runs:
+  using: composite
+  steps:
+    # Version: 3.0.2
+    - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+
+    - uses: actions/setup-node@09ba51f18e18a3756fea1f54d09c6745c064491d
+      with:
+        node-version: 16
+        cache: npm
```

Comment:
Note the use of built-in caching as opposed to actions/cache. Since this is maintained by GitHub hopefully it will work better on average than our caching has in the past.